src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -407,8 +408,14 @@ private void validateRegion(DatasetRequestModel datasetRequest, Errors errors) {
             "InvalidRegionForPlatform",
             "Cannot set a region when a cloudPlatform is not provided.");
       } else {
-        CloudPlatformWrapper.of(datasetRequest.getCloudPlatform())
-            .ensureValidRegion(datasetRequest.getRegion(), errors);
+        CloudPlatformWrapper cloudWrapper =
+            CloudPlatformWrapper.of(datasetRequest.getCloudPlatform());
+        cloudWrapper.ensureValidRegion(datasetRequest.getRegion(), errors);
+
+        if (cloudWrapper.isAzure()) {
```

Comment:
There is..it's currently broken.  In this case, we need to check the region from two different fields (note that this is checking against `datasetRequest.getGcpRegion()` which is a temporary field until the azure dataset supports being fully in azure

Reply:
OK, I see it now.

Since this is the only caller of `CloudPlatformWrapper.ensureValidRegion()` you could change it to accept `DatasetRequestModel` instead of `String` but maybe that's a messier solution than having this code here... up to you.

Reply:
Yeah, I kinda like having it up front.  And it's temporary